chia/consensus/constants.py

```diff
@@ -26,37 +28,37 @@ class ConsensusConstants:
     SUB_EPOCH_BLOCKS: uint32  # The number of blocks per sub-epoch
     EPOCH_BLOCKS: uint32  # The number of blocks per sub-epoch, must be a multiple of SUB_EPOCH_BLOCKS
 
-    SIGNIFICANT_BITS: int  # The number of bits to look at in difficulty and min iters. The rest are zeroed
-    DISCRIMINANT_SIZE_BITS: int  # Max is 1024 (based on ClassGroupElement int size)
-    NUMBER_ZERO_BITS_PLOT_FILTER: int  # H(plot id + challenge hash + signage point) must start with these many zeroes
-    MIN_PLOT_SIZE: int
-    MAX_PLOT_SIZE: int
-    SUB_SLOT_TIME_TARGET: int  # The target number of seconds per sub-slot
-    NUM_SP_INTERVALS_EXTRA: int  # The difference between signage point and infusion point (plus required_iters)
-    MAX_FUTURE_TIME: int  # The next block can have a timestamp of at most these many seconds more
-    NUMBER_OF_TIMESTAMPS: int  # Than the average of the last NUMBER_OF_TIMESTAMPS blocks
+    SIGNIFICANT_BITS: uint8  # The number of bits to look at in difficulty and min iters. The rest are zeroed
+    DISCRIMINANT_SIZE_BITS: uint16  # Max is 1024 (based on ClassGroupElement int size)
+    NUMBER_ZERO_BITS_PLOT_FILTER: uint16  # H(plot id + challenge hash + signage point) must start with these many zeroes # noqa: E501
+    MIN_PLOT_SIZE: uint16
+    MAX_PLOT_SIZE: uint16
+    SUB_SLOT_TIME_TARGET: uint32  # The target number of seconds per sub-slot
+    NUM_SP_INTERVALS_EXTRA: uint16  # The difference between signage point and infusion point (plus required_iters)
+    MAX_FUTURE_TIME: uint32  # The next block can have a timestamp of at most these many seconds more
+    NUMBER_OF_TIMESTAMPS: uint16  # Than the average of the last NUMBER_OF_TIMESTAMPS blocks
     # Used as the initial cc rc challenges, as well as first block back pointers, and first SES back pointer
     # We override this value based on the chain being run (testnet0, testnet1, mainnet, etc)
     GENESIS_CHALLENGE: bytes32
     # Forks of chia should change this value to provide replay attack protection
     AGG_SIG_ME_ADDITIONAL_DATA: bytes
     GENESIS_PRE_FARM_POOL_PUZZLE_HASH: bytes32  # The block at height must pay out to this pool puzzle hash
     GENESIS_PRE_FARM_FARMER_PUZZLE_HASH: bytes32  # The block at height must pay out to this farmer puzzle hash
-    MAX_VDF_WITNESS_SIZE: int  # The maximum number of classgroup elements within an n-wesolowski proof
+    MAX_VDF_WITNESS_SIZE: uint16  # The maximum number of classgroup elements within an n-wesolowski proof
     # Size of mempool = 10x the size of block
-    MEMPOOL_BLOCK_BUFFER: int
+    MEMPOOL_BLOCK_BUFFER: uint32
     # Max coin amount uint(1 << 64). This allows coin amounts to fit in 64 bits. This is around 18M chia.
-    MAX_COIN_AMOUNT: int
+    MAX_COIN_AMOUNT: uint64
```

Comment:
Maybe this should be uin128 to allow for higher coin amounts for custom chains.

Reply:
Hm i guess you mean forks? They should just adjust it if required, right? We assigned `uint64` here already. See https://github.com/Chia-Network/chia-blockchain/blob/00c80ae6078f2c97765fe52b2a79aef5225d76d9/chia/consensus/default_constants.py#L44